qiime_studio/api/jobs.py

```diff
@@ -147,8 +134,8 @@ def callback(future):
             job = JOBS[job_id]
 
             if results is not None:
-                for result, path in zip(results, outputs.values()):
-                    result.save(path)
+                for key, path in outputs.items():
+                    getattr(results, key).save(path)
                 job['outputs'] = {k: v.uuid for k, v in zip(outputs, results)}
```

Comment:
There isn't a public way to get the fields. I didn't want to add `.fields` to `Results` in case there was a name clash with a field named "fields". namedtuple makes all additional methods/attributes prefixed with an underscore to avoid this problem, but it makes for ugly client code.

Calling `._fields` should work, maybe we document it as a public API (similar to namedtuple)? Or we could make it public and detect a namespace clash (disallow a field called "fields"). I think I like the latter, interested in what you guys think.